File: psets/pset6/credit/credit.py
from cs50 import get_int
from cs50 import get_string

n = ""
check = 1

while (check != 0):
    check = 0
    n = get_string("Number: ")
    for i in range(len(n)):
        if (type(n[i]) == int):
            check += 1

sumhold = ""
sum1 = 0
sum2 = 0
legit = 0

if (len(n) % 2 == 0):
    for i in range(len(n)):
        if (i % 2 == 0 or i == 0):
            sumhold += str(int(n[i]) * 2)
        else:
            sum2 += int(n[i])
    for i in range(len(sumhold)):
        sum1 += int(sumhold[i])
    if ((sum1 + sum2) % 10 == 0):
        legit = 1
else:
    for i in range(len(n)):
        if (i % 2 != 0):
            sumhold += str(int(n[i]) * 2)
        else:
            sum2 += int(n[i])
    for i in range(len(sumhold)):
        sum1 += int(sumhold[i])
    if ((sum1 + sum2) % 10 == 0):
        legit = 1

f2num = int(n[0]) * 10 + int(n[1])
label = ""

if (f2num == 34 or f2num == 37):
    label = "AMEX"
elif (f2num > 50 and f2num < 56):
    label = "MASTERCARD"
elif (int(n[0]) == 4):
    label = "VISA"

if (legit == 1):
    if (len(n) == 13 and label == "VISA"):
        print("VISA")
    elif (len(n) == 15 and label == "AMEX"):
        print("AMEX")
    elif (len(n) == 16 and label == "VISA"):
        print("VISA")
    elif (len(n) == 16 and label == "MASTERCARD"):
        print("MASTERCARD")
    else:
        print("INVALID")
else:
    print("INVALID\n")

File: psets/pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int getSum(int n)
{
    int i = n;
    int sum = 0;
    int rem;

    while (i != 0)
    {
        rem = i % 10;
        sum = sum + rem;
        i = i / 10;
    }

    return sum;
}

int main(void)
{
    string n;
    int len = 0;
    int check;;
    
    do
    {
        check = 0;
        n = get_string("Number: ");
        len = strlen(n);
        for (int i = 0; i < len; i++)
        {
            if (n[i] < 48 || n[i] > 57)
            {
                check += 1;
            }
        }
    }
    while (check != 0);
    
    string comp = n;
    
    int sum = 0;
    int sum1 = 0;
    int hold = 0;
    int d1 = 0;
    int d2 = 0;
    
    if (len % 2 == 0)
    {
        for (int i = 0; i < len; i++)
        {
            hold = (n[i] - 48) * 2;;
            if (i == 0)
            {
                if (hold < 10)
                {
                    sum += (n[i] - 48) * 2;
                }
                else
                {
                    sum += getSum((n[i] - 48) * 2);
                }
            }
            else if (i % 2 == 0)
            {
                if (hold < 10)
                {
                    sum += (n[i] - 48) * 2;
                }
                else
                {
                    sum += getSum((n[i] - 48) * 2);
                }
            }
        }
        
        for (int i = 0; i < len; i++)
        {
            if (i % 2 != 0)
            {
                sum += n[i] - 48;
            }
            
        }
    }
    else
    {
        for (int i = 0; i < len; i++)
        {
            hold = (n[i] - 48) * 2;;
            if (i % 2 != 0)
            {
                if (hold < 10)
                {
                    sum += (n[i] - 48) * 2;
                }
                else
                {
                    sum += getSum((n[i] - 48) * 2);
                }
            }
        }
        
        for (int i = 0; i < len; i++)
        {
            if (i % 2 == 0)
            {
                sum += n[i] - 48;
            }
            
        }
    }
    
    if ((n[0] - 48 == 3 && (n[1] - 48 == 4 || n[1] - 48 == 7)) && sum % 10 == 0 && len == 15)
    {
        printf("AMEX\n");
    }
    else if ((n[0] - 48 == 5 && (n[1] - 48 == 1 || n[1] - 48 == 2 || n[1] - 48 == 3 || n[1] - 48 == 4 || n[1] - 48 == 5))
             && (sum % 10 == 0) && (len == 16))
    {
        printf("MASTERCARD\n");
    }
    else if (n[0] - 48 == 4 && sum % 10 == 0 && (len >= 13 && len <= 16))
    {
        printf("VISA\n");
        
    }
    else
    {
        printf("INVALID\n");
    }
}